### PR TITLE
Add a global log policy hook to the logging framework

### PR DESCRIPTION
**Pull request by ckreibich**
This addresses the need for a central hook on any log write. This is currently a pain: there's no hook management API to add an extra handler to any existing hook, and replacing the policy hook in a given log filter record requires chaining the invocation of the former one (and updating all existing filters to be truly global).

This has come up a few times since the hook introduction in 4.0, so I figured it'd be good to get into 4.1 still. If folks like this, I'll add corresponding `zeek-docs` documentation when merging.

### CODE REVIEW

File: src/logging/Manager.cc [hunk 1]
```diff
@@ -743,6 +744,15 @@ bool Manager::Write(EnumVal* id, RecordVal* columns_arg)
 				continue;
 			}
 
+		if ( global_log_policy_hook )
```
Comment (ckreibich) [inline src/logging/Manager.cc:747]: I considered iterating over the two hooks (`filter->policy` and `global_log_policy_hook`) via for-loop, but then you lose the convenient `continue` to skip the entry. So I think that bit of redundancy is okay here.

**Review by rsmmr — APPROVED** [review]
Nice extension, agree this make sense.

**Comment by ckreibich** [comment]
@rsmmr I've updated this based on our discussion. @awelzel feel free to take a look as well, thanks!

File: src/logging/Manager.cc [hunk 2]
```diff
@@ -739,7 +755,7 @@ bool Manager::Write(EnumVal* id, RecordVal* columns_arg)
 			auto v = filter->policy->Invoke(columns,
 							IntrusivePtr{NewRef{}, id},
 							IntrusivePtr{NewRef{}, filter->fval});
-			if ( v  && ! v->AsBool() )
+			if ( stream_veto || (v  && ! v->AsBool()) )
```
Comment (rsmmr) [inline src/logging/Manager.cc:758]: Should the check for `stream_veto` be outside of this `if`? Right now, if there's not filter policy hook, wouldn't the global veto be ignored?

Comment (ckreibich) [reply src/logging/Manager.cc:758]: Good catch, yep. I forgot that there may be no filters at all.

Comment (ckreibich) [reply src/logging/Manager.cc]: I added a testcase for that situation, while I was at it.

**Review by rsmmr — APPROVED** [review]

**Comment by ckreibich** [comment]
Just rewording the main commit here, since it no longer matched the functionality.

**Pull request merged**